frontend/src/main.tsx

```diff
@@ -27,8 +28,8 @@ const router = createBrowserRouter([
     element: <ProtectedRoute component={UploadPage} />,
   },
   {
-    path: "docs/:id",
-    element: <ProtectedRoute component={DocumentationPage} />,
+    path: "docs/:docType/:id",
```

Comment:
I'd highly suggest a rework of these paths:
- `/repos` -> dashboard page
- `/repos/{repo_id}` -> this is the data that populates the sidebar
- `/repos/{repo_id}/{doc_id}` -> what it currently does (sidebar + editor)

Why?
- We should be following RESTful principles like we did in the backend routes, specifically clear hierarchy.
- Our only entities are `repos` and `docs`. While a `doc` _could_ be individual, they really belong to a `repo`.
- With `/docs/:docType/:id` it's not clear what the hierarchy is, we actually are ignoring repos altogether in the URI.
- `:docType` should be a query parameter, since we don't actually have `files` or `folders` as entities, they are just a descriptor of `doc`.
  - we actually can (and should) get rid of docType altogether if we implement the paths as mentioned.
- **Most importantly**: we create consistency between the frontend and backend routes.